include/gridtools/fn/backend/common.hpp

```diff
@@ -31,10 +31,30 @@ namespace gridtools::fn::backend {
                 meta::rename<tuple, Dims>());
         }
 
+        template <class Dims, class Sizes, class UnrollFactors>
+        constexpr GT_FUNCTION auto make_unrolled_loops(Sizes const &sizes, UnrollFactors) {
+            return tuple_util::host_device::fold(
+                [&](auto outer, auto dim) {
+                    using unroll_factor = std::remove_reference_t<decltype(host_device::at_key<decltype(dim)>(
+                        std::declval<UnrollFactors const &>()))>;
```

Comment:
I am not sure, but maybe something like
```suggestion
                    using unroll_factor = std::remove_reference_t<element_at<decltype(dim), UnrollFactors>>;
```